File: assingnment_4.py
import os

def main():
    res = os.fork()

    if res == 0:
        os.execl("/bin/bash", "bash", "-c", "cat test.txt | grep test")
    
    os.wait()

    print("Parent process done")


if __name__ == "__main__":
    main()

File: concurrency/c/assingnment_4.c
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>


int main() {
    int res;
    res = fork();
    if (res == 0) {
        execl("/bin/bash", "bash", "-c", "cat test.txt | grep test", NULL);
    }

    wait(NULL);

    printf("Parent process done");

    return 0;
}
